wurst/math/Matrices.wurst

```diff
@@ -0,0 +1,310 @@
+    
+/** Returns matrix' column by id as a vector or zero-vector if doesn't exist. */
+public function mat3.col(int id) returns vec3
+    switch id
+        case 0
+            return vec3(this.rc00, this.rc10, this.rc20)
+        case 1
+            return vec3(this.rc01, this.rc11, this.rc21)
+        case 2
+            return vec3(this.rc02, this.rc12, this.rc22)
+        default
+            return ZERO3
+
+/** Returns matrix' row by id as a vector or zero-vector if doesn't exist. */
+public function mat3.row(int id) returns vec3
+    switch id
+        case 0
+            return vec3(this.rc00, this.rc01, this.rc02)
+        case 1
+            return vec3(this.rc10, this.rc11, this.rc12)
+        case 2
+            return vec3(this.rc20, this.rc21, this.rc22)
+        default
+            return ZERO3
+        
+public function mat3.determinant() returns real
+    return this.rc00*(this.rc11*this.rc22 - this.rc12*this.rc21)
+         - this.rc01*(this.rc10*this.rc22 - this.rc12*this.rc20)
+         + this.rc02*(this.rc10*this.rc21 - this.rc11*this.rc20)
+
+/** Flips the matrix over its main diagonale.
+Result of transposing for rotation matrix is its inverse matrix. */
+public function mat3.transpose() returns mat3
+    return mat3(
+        this.rc00, this.rc10, this.rc20,
+        this.rc01, this.rc11, this.rc21,
+        this.rc02, this.rc12, this.rc22)
+
+/** Returns a zero-matrix in case there's no iverse matrix.*/
+public function mat3.inverse() returns mat3
+    var d = this.determinant()
+    if d.abs().isBetween(0.998, 1.002)
+        return this.transpose()
+    else if d.abs() > 0.002
+        return 1/d*mat3(
+            this.rc11*this.rc22 - this.rc12*this.rc21,
+            this.rc02*this.rc21 - this.rc01*this.rc22,
+            this.rc01*this.rc12 - this.rc02*this.rc11,
+        
+            this.rc12*this.rc20 - this.rc10*this.rc22,
+            this.rc00*this.rc22 - this.rc20*this.rc02,
+            this.rc02*this.rc10 - this.rc00*this.rc12,
+
+            this.rc10*this.rc21 - this.rc11*this.rc20,
+            this.rc01*this.rc20 - this.rc00*this.rc21,
+            this.rc00*this.rc11 - this.rc01*this.rc10)
+    return ZERO33
+
+/** Extracts Euler-angles (Tait-Bryan) from rotation matrix. Order of result angles is X-Y-Z.*/
+public function mat3.toEuler() returns vec3
+    real yaw
+    real pitch = Asin(this.rc02)
+    real roll
+    var cosY = Cos(pitch)
+    if cosY.abs() > 0.005
+        var cosX = this.rc22/cosY
+        var sinX = -this.rc12/cosY
+        yaw  = Atan2(sinX, cosX)
+        var cosZ =  this.rc00/cosY
+        var sinZ = -this.rc01/cosY
+        roll = Atan2(sinZ, cosZ)
+    else
+        yaw = 0.
+        var cosZ = this.rc11
+        var sinZ = this.rc10
+        roll  = Atan2(sinZ, cosZ) 
+    return vec3(yaw, pitch, roll)
+
+public function mat3.toString() returns string
+    return "Matrix 3x3\n{0} {1} {2}\n{3} {4} {5}\n{6} {7} {8}".format(

```

Comment:
Don't use .format(), it's very infefficient.

Reply:
Is performance above readability in such debug functions?